test/TestCaseReader.cpp

```diff
@@ -156,7 +200,7 @@ pair<SourceMap, size_t> TestCaseReader::parseSourcesAndSettingsWithLineNumber(is
 	}
 	// Register the last source as the main one
 	sources[currentSourceName] = currentSource;
-	return {{move(sources), move(currentSourceName)}, lineNumber};
+	return {{move(sources), move(externalSources), move(currentSourceName)}, lineNumber};
```

Comment:
Why do we have to distinguish between sources and external sources?

Reply:
I distinguish sources and external sources to enable the correct generation of expectation updates.

Reply:
Also `sources` contains external sources. It holds also the content of external sources. I didn't wanted to touch the compilation logic of present in `SolidityExecutionFramework`. `sources` and `externalSources` use the same keys, where the value of `source` points to it's content. For `externalSources` it points to the path of that external source. This information is only used to update expectations.